### ISSUE TITLE
Setting HTMX Headers does not work when using HtmxView

### ISSUE THREAD

**Issue by frederikb** [wimdeblauwe/htmx-spring-boot#150]
Hi, thanks for sharing this library. Good to see some tooling around JTE and HTMX. :)

I've upgraded to 3.6.1 and tried to migrate to the new recommended approach, which is using `HtmxView` as a response type and using `HtmxResponse` as a parameter. Unfortunately using `HtmxView` seems to break setting HTMX headers via `HtmxResponse` and additionally for annotations when used for an @ExceptionHandler.

Of course this could be related to my setup (Spring Boot 3.3.5, JTE 3.1.13).

What works:

```java
// TC1
@PostMapping
@HxRequest
@HxReswap(HxSwapType.NONE)
public String view(Model model) {
  return "view";
}

// TC2
@PostMapping
@HxRequest
public String view(Model model, HtmxResponse response) {
  response.setReswap(HtmxReswap.none());
  return "view";
}

// TC3
@PostMapping
@HxRequest
@HxReswap(HxSwapType.NONE)
public HtmxView view(Model model) {
  return new HtmxView("view");
}
```

Does not work:

```java
// TC4
@PostMapping
@HxRequest
public HtmxView view(Model model, HtmxResponse response) {
  response.setReswap(HtmxReswap.none());
  return new HtmxView("view");
}

// TC5
@ExceptionHandler(Exception.class)
@HxReswap(HxSwapType.NONE)
public HtmxView handleException(Exception ex) {
  return new HtmxView("error");
}

// TC6
@ExceptionHandler(Exception.class)
public HtmxView handleException(Exception ex, HtmxResponse response) {
  response.setReswap(HtmxReswap.none());
  return new HtmxView("error");
}
```

I've briefly analyzed it by stepping through the code and the issue is as follows:

At the time that we set the headers (either via `HtmxHandlerInterceptor` for the `HtmxResponse` parameter variant or `HtmxViewMethodReturnValueHandler` for annotations) the `Response` is already committed, thus setting new headers fails silently.

The commit happens as a result of the following call:
https://github.com/wimdeblauwe/htmx-spring-boot/blob/9b58a112193441d81096a7893c9e3700dd144c26/htmx-spring-boot/src/main/java/io/github/wimdeblauwe/htmx/spring/boot/mvc/HtmxViewMethodReturnValueHandler.java#L87

Why exactly this happens I do not fully understand, but I'm not to familiar with the internal here. I would have expected that commit only happens after a flush, which I'm not seeing.

In any case this affects setting headers if they are processed after this line of code was run.

Thus my question:
- is this supposed to work like this or is my code perhaps wrong?
- does it work for anyone else which would suggest an conflicting configuration/environment on my end

Thanks! Hope this is clear.

**Comment by wimdeblauwe** [comment]
You give 3 examples of things that do not work. But the first example does not compile. Given you return a `HtmxView` but the method return type is `String`. Maybe that example is wrong? I tried the 2nd example and that indeed does not seem to work.Also, is the 3rd example correct? I don't see any setting of response headers there?

**Comment by wimdeblauwe** [comment]
What I have found so far. This works (It is part of our test suite):

```
        @ExceptionHandler(RuntimeException.class)
        public String handleError(RuntimeException ex, HtmxRequest htmxRequest, HtmxResponse htmxResponse) {
            if (htmxRequest.isHtmxRequest()) {
                htmxResponse.setRetarget("#container");
            }
            return "view1";
        }
```

However, this does not work:

```
        @ExceptionHandler(RuntimeException.class)
        @HxRetarget("#container")
        public String handleError(RuntimeException ex, HtmxRequest htmxRequest, HtmxResponse htmxResponse) {
            return "view1";
        }
```

The annotation that is put on the `handleError` method is not read by `HtmxHandlerInterceptor` as the `preHandle` method checks the controller method that throws the exception, not the method that "catches" the exception.
If I change this:

```
        @HxRequest
        @GetMapping("/throw-exception")
        public void throwException() {
            throw new RuntimeException();
        }
```

to

```
        @HxRequest
        @GetMapping("/throw-exception")
        @HxRetarget("#container")
        public void throwException() {
            throw new RuntimeException();
        }
```

Then the response header is correctly set. However, that is probably not really how we want it.

From a functional standpoint, I guess users would want to be able to set (or override!) the response headers where the exception is handled.

**Comment by wimdeblauwe** [comment]
I have added a test called `testExceptionHandlerUsingAnnotation()` on branch `feature/gh-150` that shows the problem.

**Comment by wimdeblauwe** [comment]
I have now pushed a fix on branch `feature/gh-150`. Would love for others to take a look if it is the best way to fix it.

Still todo:

* Update README to show how it works
* Add test where the exception handler overwrites the annotation present on the controller method.
* Test if it works if the exception handler is part of a `@ControllerAdvice`.

**Comment by wimdeblauwe** [comment]
I believe all your examples are fixed now. Is there a chance you can use the code on branch `feature/gh-150` and run `mvn install` locally? Then set the version to `3.6.2-SNAPSHOT` on your project and test if everything is fixed now for you @frederikb ?

**Comment by frederikb** [comment]
Thanks @wimdeblauwe, that was quick! I'll be happy to test out the branch on Monday and report back.

**Comment by xhaggi** [comment]
@wimdeblauwe I've had some time to look at your implementation and I've simplified it a bit. https://github.com/wimdeblauwe/htmx-spring-boot/compare/main...xhaggi:fix/gh-150.
Mind taking a look? Let me know if everything looks good and I'll create a PR and do a backport once it's merged.

**Comment by wimdeblauwe** [comment]
@xhaggi Amazing work, that is indeed a lot better!

**Comment by frederikb** [comment]
Hi @wimdeblauwe and @xhaggi 

thanks for the quick work on this. I tested it with the branch from @xhaggi:

-  ✅  TC5 and TC6
- ❌ TC4 (which is not related to error handling)

Did you test TC4 and were able to confirm it and was that intended to be fixed as well?

**Comment by xhaggi** [comment]
@frederikb I did a quick test and it works.

**Comment by wimdeblauwe** [comment]
@frederikb I have now released 3.6.2: https://github.com/wimdeblauwe/htmx-spring-boot/releases/tag/3.6.2
